server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4643,12 +4643,14 @@ public void syncSegmentsFromRemoteSegmentStore(boolean overrideLocal, boolean re
                         indexInput,
                         remoteSegmentMetadata.getGeneration()
                     );
-                    if (shouldCommit) {
+                    // Replicas never need a local commit
+                    if (shouldCommit && this.shardRouting.primary()) {
```

Comment:
As I recall, `shouldCommit` flag was introduced with remote store and replication integration. If we are adding a check on primary, do we need a separate `shouldCommit` flag?

Reply:
This can be taken in a follow-up refactor along with testing for primary relocation.